### PR TITLE
NodeJS: shorter paragraphs and more visible examples fix

### LINKED ISSUE #514: NodeJS: shorter paragraphs and more visible examples
### Description
Currently we show the whole description for each article, which results in a cluttered and often too wordy answer, and the examples usually are too far down and require the user to expand the content. We should instead show the first sentence and then the examples when available.

screen shot 2016-11-03 at 2 21 59 pm

## Get Started
- [ ] 1) Claim this issue by commenting below
- [ ] 2) Review our [Contributing Guide](https://github.com/duckduckgo/zeroclickinfo-fathead/blob/master/CONTRIBUTING.md)
- [ ] 3) [Set up your development environment](https://docs.duckduckhack.com/welcome/setup-dev-environment.html), and fork this repository
- [ ] 4) Create a Pull Request

## Resources
- Join [DuckDuckHack Slack](https://quackslack.herokuapp.com/) to ask questions
- Join the [DuckDuckHack Forum](https://forum.duckduckhack.com/) to discuss project planning and Instant Answer metrics
- Read the [DuckDuckHack Documentation](https://docs.duckduckhack.com/) for technical help

---

Instant Answer Page: https://duck.co/ia/view/node_js

### PR DESCRIPTION
**Pull request by ahetawal-p**
## Description of new Instant Answer, or changes
Added a new processor method used for extracting only the relevant description content. This change always gets the first p tag for a field Name. Since examples are what users are looking for in a fathead, rest of the abstract content just fills in all the pre tags which are: "example" section in NodeJs documentation. For keeping some contextual info, also adding a previous p tag for each example pre tag in the final content, so as not to break the documentation continuity

## Related Issues and Discussions
Fixes #514

## People to notify
@MariagraziaAlastra @tagawa @moollaza 

## Testing & Review
To be completed by Language Leader (or DDG Staff)

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page**
- [ ] Instant Answer page is correctly filled out and contains:
    - One topic for the Search Space Language (Java, Python, Scala, Ruby, etc.)
    - One topic from: Reference, Help, Libraries, Tools
        - Documentation Fatheads are considered "Reference"
    - Description, Data source, and 2+ example queries
    - Perl Module (e.g. "DDG::Fathead::PerlDoc" -- we only need a name, not an actual file)
    - Source Name (for "More at <source_name>" link)
    - Source Domain (must contain http:// or https:// -- can be the same as Data Source)
    - Source Info (used as Subtitle for each Article -- usually matches the IA Name)
    - 'Skip Abstract' is checked off
    - Source ID (ping @moollaza to assign one, once Fathead is ready for Beta deploy)
- [ ] Fathead Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Code**
- [x] Uniformly indented, well commented
- [x] Fetch.sh and Parse.sh run without errors
- [x] Output contains no blank lines, or multi-line entries

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/node_js

### CODE REVIEW

**Comment by daxtheduck** [comment]
@ahetawal-p Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/node_js/parse.js [hunk 1]
```diff
@@ -30,12 +31,13 @@ function createItem(ops) {
 }
 
 function getAbstract(obj) {
-    var a = obj.desc ? '<p id="obj.desc">' + obj.desc + '</p>' : '',
+    var summarizedDesc = obj.desc ? preProcessDescription(obj.desc) : '';
+    var a =  summarizedDesc,
         code = '',
-        a = a.replace(/<\s*p*>/ig, ''),
+     // a = a.replace(/<\s*p*>/i, ''),
```
Comment (bfmags) [inline lib/fathead/node_js/parse.js]: @ahetawal-p Looks good, but we should remove any commented out code from the PR (we can always retrieve it through Git). Cheers!

**Comment by bfmags** [comment]
@ahetawal-p Well done! This IA looks much better 👍

**Comment by ahetawal-p** [comment]
Hey @bfmags updated the PR after code review comments. Let me know if its good for merging.

**Comment by bfmags** [comment]
hey @ahetawal-p, unfortunately found a lot of duplicates in the output.txt 

* Running tests in duckpan: 
`DDG_TEST_FATHEAD=node_js prove --quiet t/validate_fathead.t 1>/dev/null`
* Results:
https://gist.github.com/bfmags/86f81caf513c0078b1d3be3918c22526

**Comment by ahetawal-p** [comment]
hmm, I didnt touch any of the keys generation part. I thought removing was done in a previous PR. I can take a look but should be a different PR, as it might be some other part of code which is affecting this.

**Comment by ahetawal-p** [comment]
@bfmags do you think we should merge this one, and open another bug for the the duplicates ?

**Comment by bfmags** [comment]
@ahetawal-p yes, just confirmed it was there before! I'll merge this one and open a new issue for the dups! 👍 

Thanks!
